src/drivers/fpga/socfpga/socfpga.c

```diff
@@ -0,0 +1,355 @@
+	 * what is being done here */
+	static const int (*fpga2sdram_apply)(void) = (void *) 0x3ff7d52c;
+	REG32_STORE(SOCFPGA_SYSMGR_INTF, 0);
+	fpga2sdram_apply();
+	REG32_STORE(SOCFPGA_SDRAM_CONFIG, 0);
+	REG32_STORE(SOCFPGA_RESETMGR_AXI_BRIDGE, 0);
+	REG32_STORE(SOCFPGA_L3_REMAP, 0x19);
+#endif
+}
+
+struct socfpga {
+	uintptr_t mgr_base;
+	uintptr_t mgr_data_addr;
+};
+
+static void socfpga_or(struct socfpga *socfpga, int offt, uint32_t val) {
+	uint32_t tmp = REG32_LOAD(socfpga->mgr_base + offt);
+
+	REG32_STORE(socfpga->mgr_base + offt, val | tmp);
+}
+
+static void socfpga_clear(struct socfpga *socfpga, int offt, uint32_t val) {
+	uint32_t tmp = REG32_LOAD(socfpga->mgr_base + offt);
+
+	REG32_STORE(socfpga->mgr_base + offt, tmp & ~val);
+}
+
+static uint32_t socfpga_read(struct socfpga *socfpga, int offt) {
+	return REG32_LOAD(socfpga->mgr_base + offt);
+}
+
+static void socfpga_write(struct socfpga *socfpga, int offt, uint32_t val) {
+	REG32_STORE(socfpga->mgr_base + offt, val);
+}
+
+static void socfpga_data_write(struct socfpga *socfpga, uint32_t val) {
+	REG32_STORE(socfpga->mgr_data_addr, val);
+}
+
+static void socfpga_regdump(struct socfpga *socfpga) {
+	log_debug(" STAT             %08x", socfpga_read(socfpga, SOCFPGA_MGR_STAT));
+	log_debug(" CTRL             %08x", socfpga_read(socfpga, SOCFPGA_MGR_CTRL));
+	log_debug(" DCLKCNT          %08x", socfpga_read(socfpga, SOCFPGA_MGR_DCLKCNT));
+	log_debug(" DCLKSTAT         %08x", socfpga_read(socfpga, SOCFPGA_MGR_DCLKSTAT));
+	log_debug(" GPO              %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPO));
+	log_debug(" GPI              %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPI));
+	log_debug(" MISCI            %08x", socfpga_read(socfpga, SOCFPGA_MGR_MISCI));
+	log_debug(" GPIO_INTEN       %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_INTEN));
+	log_debug(" GPIO_INTMASK     %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_INTMASK));
+	log_debug(" GPIO_INTTYPE_LEV %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_INTTYPE_LEVEL));
+	log_debug(" GPIO_INT_POLARIT %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_INT_POLARITY));
+	log_debug(" GPIO_INTSTATUS   %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_INTSTATUS));
+	log_debug(" GPIO_RAW_INTSTAT %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_RAW_INTSTATUS));
+	log_debug(" GPIO_PORTA_EOI   %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_PORTA_EOI));
+	log_debug(" GPIO_EXT_PORTA   %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_EXT_PORTA));
+	log_debug(" GPIO_LS_SYNC     %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_LS_SYNC));
+	log_debug(" GPIO_VER_ID_CODE %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_VER_ID_CODE));
+	log_debug(" GPIO_CONFIG_REG2 %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_CONFIG_REG2));
+	log_debug(" GPIO_CONFIG_REG1 %08x", socfpga_read(socfpga, SOCFPGA_MGR_GPIO_CONFIG_REG1));
+}
+
+static uint32_t socfpga_mode(struct socfpga *socfpga) {
+	uint32_t ret = socfpga_read(socfpga,
```

Comment:
May be separate a declaration and using?

Reply:
I don't think that it makes much difference as we have a single variable here